dot/parachain/candidate-validation/host_test.go

```diff
@@ -0,0 +1,326 @@
+		})
+	}
+}
+
+func TestHost_performBasicChecks(t *testing.T) {
+	paramsTooLarge := ParamsTooLarge
+	povHashMismatch := PoVHashMismatch
+	codeHashMismatch := CodeHashMismatch
+	badSignature := BadSignature
+
+	pov := parachaintypes.PoV{
+		BlockData: []byte{1, 2, 3, 4, 5, 6, 7, 8},
+	}
+	povHash, err := pov.Hash()
+	pov2 := parachaintypes.PoV{
+		BlockData: []byte{1, 1, 1, 1, 1},
+	}
+	validationCode := parachaintypes.ValidationCode{1, 2, 3}
+	validationCodeHash := validationCode.Hash()
+
+	require.NoError(t, err)
+	collatorKeypair, err := sr25519.GenerateKeypair()
+	require.NoError(t, err)
+	collatorID, err := sr25519.NewPublicKey(collatorKeypair.Public().Encode())
+	require.NoError(t, err)
+
+	candidate := parachaintypes.CandidateDescriptor{
+		Collator:           collatorID.AsBytes(),
+		PovHash:            povHash,
+		ValidationCodeHash: validationCodeHash,
+	}
+	candidate2 := candidate
+
+	payload, err := candidate.CreateSignaturePayload()
+	require.NoError(t, err)
+
+	signatureBytes, err := collatorKeypair.Sign(payload)
+	require.NoError(t, err)
+
+	signature := [sr25519.SignatureLength]byte{}
+	copy(signature[:], signatureBytes)
+
+	signature2Bytes, err := collatorKeypair.Sign([]byte{1, 2, 3, 4, 5, 6, 7, 8})
+	require.NoError(t, err)
+	signature2 := [sr25519.SignatureLength]byte{}
+	copy(signature2[:], signature2Bytes)
+
+	candidate.Signature = parachaintypes.CollatorSignature(signature)
+	candidate2.Signature = parachaintypes.CollatorSignature(signature2)
+
+	type args struct {
+		candidate          *parachaintypes.CandidateDescriptor
+		maxPoVSize         uint32
+		pov                parachaintypes.PoV
+		validationCodeHash parachaintypes.ValidationCodeHash
+	}
+	tests := map[string]struct {
+		args          args
+		expectedError *ReasonForInvalidity
+	}{
+		"params_too_large": {
+			args: args{
+				candidate:  &candidate,
+				maxPoVSize: 2,
+				pov:        pov,
+			},
+			expectedError: &paramsTooLarge,
+		},
+		"invalid_pov_hash": {
+			args: args{
+				candidate:  &candidate,
+				maxPoVSize: 1024,
+				pov:        pov2,
+			},
+			expectedError: &povHashMismatch,
+		},
+		"invalid_code_hash": {
+			args: args{
+				candidate:          &candidate,
+				maxPoVSize:         1024,
+				pov:                pov,
+				validationCodeHash: parachaintypes.ValidationCodeHash{1, 2, 3},
+			},
+			expectedError: &codeHashMismatch,
+		},
+		"invalid_signature": {
+			args: args{
+				candidate:          &candidate2,
+				maxPoVSize:         1024,
+				pov:                pov,
+				validationCodeHash: validationCodeHash,
+			},
+			expectedError: &badSignature,
+		},
+		"happy_path": {
+			args: args{
+				candidate:          &candidate,
+				maxPoVSize:         1024,
+				pov:                pov,
+				validationCodeHash: validationCodeHash,
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			validationError, _ := performBasicChecks(tt.args.candidate, tt.args.maxPoVSize, tt.args.pov,
```

Comment:
Can we run these cases parallelly?
Also, Why do you ignore internal error?

Reply:
Added parallel to tests and added check for internal errors.